oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/postgresql/PostgreSQLTableInstaller.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.jdbc.postgresql;
+
+import com.google.gson.JsonObject;
+import org.apache.skywalking.oap.server.core.analysis.NodeType;
+import org.apache.skywalking.oap.server.core.storage.model.ModelColumn;
+import org.apache.skywalking.oap.server.core.storage.type.StorageDataComplexObject;
+import org.apache.skywalking.oap.server.library.client.Client;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.mysql.MySQLTableInstaller;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+
+public class PostgreSQLTableInstaller extends MySQLTableInstaller {
+
+    public PostgreSQLTableInstaller(Client client, ModuleManager moduleManager, int maxSizeOfArrayColumn,
+                                    int numOfSearchableValuesPerTag) {
+        super(client, moduleManager, maxSizeOfArrayColumn, numOfSearchableValuesPerTag);
+    }
+
+    protected String transform(ModelColumn column, Class<?> type, Type genericType) {
+        final String storageName = column.getColumnName().getStorageName();
+        if (Integer.class.equals(type) || int.class.equals(type) || NodeType.class.equals(type)) {
+            return storageName + " INT";
+        } else if (Long.class.equals(type) || long.class.equals(type)) {
+            return storageName + " BIGINT";
+        } else if (Double.class.equals(type) || double.class.equals(type)) {
+            return storageName + " DOUBLE PRECISION";
+        } else if (String.class.equals(type)) {
+            return storageName + " VARCHAR(" + column.getLength() + ")";
+        } else if (StorageDataComplexObject.class.isAssignableFrom(type)) {
+            return storageName + " VARCHAR(20000)";
```

Comment:
Could you use switch statement instead of if-else statement?

Reply:
OK, I noticed `org.apache.skywalking.oap.server.storage.plugin.jdbc.h2.dao.H2TableInstaller#transform` it is the same, do I need to change it together?

Reply:
More difficult, the `type` cannot be used in switch expressions.

Reply:
I thought in the beginning, using `switch(type.getTypeName())`. But now, I think this way is okay.